sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubIO.java

```diff
@@ -735,6 +735,20 @@ public static Write<PubsubMessage> writeMessages() {
         .build();
   }
 
+  /**
+   * Returns A {@link PTransform} that writes {@link PubsubMessage}s, along with the {@link
+   * PubsubMessage#getMessageId() messageId} and {@link PubsubMessage#getOrderingKey()}, to a Google
+   * Cloud Pub/Sub stream.
+   */
+  public static Write<PubsubMessage> writeMessagesWithOrderingKey() {
+    return Write.newBuilder()
+        .setTopicProvider(null)
+        .setTopicFunction(null)
+        .setDynamicDestinations(false)
+        .setNeedsOrderingKey(true)
+        .build();
+  }
+
```

Comment:
A non-static method `withOrderingKey()` should be fine since this can function independent of dynamic destinations.
It would have no effect to call that method on a `Write` constructed using `writeStrings()`, `writeProtos()` or `writeAvros()`, but that's also the case for `withTimestampAttribute()` and `withIdAttribute()`.